### PR TITLE
Update redux-form to 6.2.x

### PR DESCRIPTION
**Pull request by LKay**
Please fill in this template.

- [ ] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [X] Run `npm run lint -- package-name` if a `tslint.json` is present.

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Run `tsc` without errors.
- [ ] Include the required [files](https://github.com/DefinitelyTyped/DefinitelyTyped#create-a-new-package) and header. Base these on the README, *not* on an existing project.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [X] Increase the version number in the header if appropriate.

### CODE REVIEW

File: redux-form/index.d.ts [hunk 1]
```diff
@@ -1,476 +1,403 @@
...
+    const getFormValues: DataSelector;
+    const getFormSyncErrors: ErrorsSelector;
+    const getFormSubmitErrors: ErrorsSelector;
+    const isDirty: BooleanSelector;
+    const isPristine: BooleanSelector;
+    const isValid: BooleanSelector;
+    const isInvalid: BooleanSelector;
 }
 
+export as namespace ReduxForm
```
Comment (aikoven) [inline redux-form/index.d.ts]: You don't have to declare a root namespace when you use `export as namespace`. Just use regular `export`s on top level

File: redux-form/index.d.ts [hunk 2]
```diff
@@ -1,476 +1,403 @@
...
+
+    interface Config<FormData extends DataShape, P, S> {
+        form: string;
+        asyncBlurFields?: string[];
+        asyncValidate?(values: FormData, dispatch: Redux.Dispatch<S>, props: P, blurredField?: string): Promise<any & FormError & FormWarning>;
+        destroyOnUnmount?: boolean;
+        enableReinitialize?: boolean;
+        getFormState?: FromStateGetter<S>;
+        keepDirtyOnReinitialize?: boolean;
+        initialValues?: PartialData<FormData>;
```
Comment (aikoven) [inline redux-form/index.d.ts]: Why not just use built in `Partial<FormData>`?

**Comment by aikoven** [comment]
@LKay Nice work! I don't use `redux-form` in my current projects, so I can't say anything about the correctness of API typings. My only concern is that you are using an old-style definition format, i.e. declaring everything inside a namespace and then exporting the whole namespace via `export =`. The modern way is to write definition as ES module and then add `export as namespace` for UMD builds.

File: redux-form/index.d.ts (hunk 1 above)
Comment (LKay) [inline redux-form/index.d.ts]: You're right. I'll change than tomorrow.

**Comment by CarsonF** [comment]
This duplicates my work and I already have a PR here #13214

**Comment by LKay** [comment]
@CarsonF Haven't seen that PR of yours. In that case, I'll just close this one and wait for yours to be merged.

**Comment by CarsonF** [comment]
@LKay Thanks :) Looks like you have some good stuff with `keyof` usage. Maybe that can be merged on top of mine?

**Pull request closed without merging**